File: src/main/java/ru/mdashlw/hypixel/ranked/reply/impl/LeaderboardReply.java
package ru.mdashlw.hypixel.ranked.reply.impl;

import com.fasterxml.jackson.annotation.JsonProperty;
import ru.mdashlw.hypixel.ranked.entities.LeaderboardPlayer;
import ru.mdashlw.hypixel.ranked.reply.Reply;

import java.util.List;

public class LeaderboardReply extends Reply<List<LeaderboardPlayer>> {
    @JsonProperty
    private List<LeaderboardPlayer> leaderboard;

    @Override
    public List<LeaderboardPlayer> getValue() {
        return leaderboard;
    }
}

File: src/main/kotlin/ru/mdashlw/hypixel/api/ranked/reply/impl/LeaderboardReply.kt
package ru.mdashlw.hypixel.api.ranked.reply.impl

import ru.mdashlw.hypixel.api.ranked.elements.Leaderboard
import ru.mdashlw.hypixel.api.ranked.reply.Reply

class LeaderboardReply(
    success: Boolean,
    cause: String?,
    leaderboard: Leaderboard?
) : Reply<Leaderboard>(success, cause, leaderboard)
